homepage/src/content/blog/ts-enums/typescript-enums.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Why you shouldn't use Enums!
+description: TypeScript Enums are heavily overused in my opinion. Here are plenty of reasons why!
+pubDate: 2024-03-29
+heroImage: ./heroImage.png
+heroImageAlt: A hand holding the Typescript logo with 'Enums' Label being thrown in a trash can
+public: true
+slug: typescript-enums
+socialMediaImage: ./heroImageTwitter.png
+
+---
+
+TypeScript `Enums` is a feature that gets way more love from the community than it actually deserves. While TypeScript 5 improved some of my original concerns,
+`Enums` are still having one major issue... There is no JavaScript concept of an `Enum`. This doesn't sound like a big deal, right? Wrong! Let's take a closer look
+at the issues and how to fix `Enums`!
+
+## The Problem
+
+Alright `Enums`don't exist in JavaScript, what's the big deal, right?! We all heard that statement 'TypeScript is a superset of JavaScript', so why shouldn't
+that superset add features to JavaScript. Well ... let me tell you the story of `experimentalDecorators`. In times not too long ago, there was just a TypeScript
+poor little compiler flag called `experimentalDecorators`. This flag was trying to act and behave like his big brother the TC39 `Decorator` Proposal.
+But unfortunately the `Decorator` proposal wasn’t all grown up yet, so as it happens to brothers, they grow apart. End of the story.  
+TypeScript introduced Decorators, modeled after the TC39 proposal as it was designed at that time, but it happened that the proposal changed while the `experimentalDecorator` feature is used and therefore can't easily
+be changed.
+To cope with the fact that there is not an `enum` construct in JavaScript, TypeScript came up with a clever trick.
+
+![Showing the code of a TS Enum defining some HTTP Status Code and also the resulting JS code](./ts-enums.png)
+
+A little simplified but basically TypeScript creates an Object that looks like this
+
+```ts
+export const HttpStatusCode = {
+    OK: 200,
+    BAD_REQUEST: 400,
+    UNAUTHORIZED: 401,
+    FORBIDDEN: 403,
+    NOT_FOUND: 404,
+    INTERNAL_SERVER_ERROR: 500,
+    200: "OK",
+    400: "BAD_REQUEST",
+    403: "FORBIDDEN",
+    404: "NOT_FOUND",
+    500: "INTERNAL_SERVER_ERROR"
+};
+```
+
+The nice thing with this approach is that is let's us have access to every possible information of the `HttpStatusCode` in a biderectional way. We can get results by
+calling `HttpStatusCode["OK"]` & `HttpStatusCode[200]`. From a developer experience this is nice and opens the door for a lot of different ways to get access to these informations,
+but it does come an increase in bundle size. That bundle size might be negligble, but it can easily add up, just imagine if I would have added all possible values
+to my `HttpStatusCode` Enum, I just picked some common one, but just have a quick look at the [https://en.wikipedia.org/wiki/List_of_HTTP_status_codes](full%20list), and it should give you a rough impression what it does
```

Comment:
I'd also change to a period here: "...Enum. I just..."

Reply:
The url wasn't replaced with "full list".

![Screenshot 2024-03-28 at 12 50 18 PM](https://github.com/niklas-wortmann/personal-site/assets/6683747/d75334d9-494c-497d-902f-2a93bb078af8)